src/libraries/System.Reflection.Emit/tests/PersistedAssemblyBuilder/AssemblySaveILGeneratorTests.cs

```diff
@@ -2532,5 +2532,78 @@ public void ConstructorOfGenericTypeReferencedCorrectly()
                 tlc.Unload();
             }
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotMonoRuntime))]
+        public void TypeBuilderArrayReferencedInIL()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                PersistedAssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderAndTypeBuilder(out TypeBuilder typeBuilder);
+                typeBuilder.DefineDefaultConstructor(MethodAttributes.Public);
+                MethodBuilder mb = typeBuilder.DefineMethod("Break", MethodAttributes.Static | MethodAttributes.Public, typeof(bool), [typeof(object)]);
+                ILGenerator il = mb.GetILGenerator();
+                il.Emit(OpCodes.Ldarg_0);
+                il.Emit(OpCodes.Isinst, typeBuilder.MakeArrayType());
+                il.Emit(OpCodes.Ret);
```

Comment:
This is invalid IL. The method should either return `object` instead of bool; or there needs to be `ldnull` + `cgt.un` here to convert the object to bool.

Once this is fixed, I expect that the test is going to pass on Mono.

Reply:
Sorry, that's on me.  That was copied directly from my original test; I wrote that intuitively (an "is instance" opcode feels like it should be an `is`, not an `as`) rather than looking up the formal specification.  Good catch!